File: packages/native-android/wrst-runtime/src/main/java/com/wrst/runtime/AssetResolver.kt
package com.wrst.runtime

// Resolves an Image `src` to a Coil model string:
//   - a full URL (contains "://")            → used as-is (remote image)
//   - a project-local name, in debug         → http://<devhost>:8081/assets/<name>
//                                               (served live by the dev server)
//   - a project-local name, in release       → file:///android_asset/wrst-assets/<name>
//                                               (embedded by `wrst build-android`)
// iOS twin: AssetResolver.swift.
object AssetResolver {
    @Volatile
    var debug = false

    fun resolve(src: String): String {
        if (src.contains("://")) return src
        return if (debug) {
            "http://$EMULATOR_HOST_IP:8081/assets/$src"
        } else {
            "file:///android_asset/wrst-assets/$src"
        }
    }
}

File: packages/native-ios/wrst-runtime/Sources/WrstRuntime/AssetResolver.swift
import Foundation

// Where an Image's `src` should load from. Android twin: AssetResolver.kt.
enum AssetSource {
    case url(URL)           // remote, or the dev server's /assets URL in debug
    case bundled(String)    // a resource embedded by `wrst build-ios` (release)
    case none
}

enum AssetResolver {
    // - full URL (contains "://")        → used as-is (remote image)
    // - project-local name, debug        → http://localhost:8081/assets/<name> (dev server)
    // - project-local name, release      → bundled resource (embedded at build time)
    static func resolve(_ src: String) -> AssetSource {
        if src.contains("://") {
            return URL(string: src).map { .url($0) } ?? .none
        }
        #if DEBUG
        return URL(string: "http://localhost:8081/assets/\(src)").map { .url($0) } ?? .none
        #else
        return .bundled(src)
        #endif
    }
}
